scripts/deploy/32_eth_autocall_vault.ts

```diff
@@ -0,0 +1,209 @@
+    await getNamedAccounts();
+  console.log(`32 - Deploying ETH Autocall Vault on ${network.name}`);
+
+  const chainId = network.config.chainId;
+  if (chainId !== CHAINID.ETH_MAINNET) {
+    console.log(`Error: chainId ${chainId} not supported`);
+    return;
+  }
+
+  const manualVolOracle = await deployments.get("ManualVolOracle");
+  const underlyingOracle = ETH_PRICE_ORACLE[chainId];
+  const stablesOracle = USDC_PRICE_ORACLE[chainId];
+
+  const manualVolOracleContract = await ethers.getContractAt(
+    ManualVolOracle_ABI,
+    manualVolOracle.address
+  );
+  const optionId = await manualVolOracleContract.getOptionId(
+    getDeltaStep("WETH"),
+    WETH_ADDRESS[chainId],
+    USDC_ADDRESS[chainId],
+    true
+  );
+
+  const pricer = await deploy("OptionsPremiumPricerWETH", {
+    from: deployer,
+    contract: {
+      abi: OptionsPremiumPricerInStables_ABI,
+      bytecode: OptionsPremiumPricerInStables_BYTECODE,
+    },
+    args: [optionId, manualVolOracle.address, underlyingOracle, stablesOracle],
+  });
+
+  console.log(`AutocallVaultWETH pricer @ ${pricer.address}`);
+
+  // Can't verify pricer because it's compiled with 0.7.3
+
+  const strikeSelection = await deploy("ManualStrikeSelectionAutocall", {
+    contract: "ManualStrikeSelection",
+    from: deployer,
+    args: [],
+  });
+
+  console.log(`AutocallVaultWETH strikeSelection @ ${strikeSelection.address}`);
+
+  try {
+    await run("verify:verify", {
+      address: strikeSelection.address,
+      constructorArguments: [],
+    });
+  } catch (error) {
+    console.log(error);
+  }
+
+  const lifecycleTreasury = await deploy("VaultLifecycleTreasury", {
+    contract: "VaultLifecycleTreasury",
+    from: deployer,
+  });
+  console.log(`VaultLifeCycleTreasury @ ${lifecycleTreasury.address}`);
+
+  // deploy logic
+  const logicDeployment = await deploy("RibbonAutocallVault", {
+    contract: "RibbonAutocallVault",
+    from: deployer,
+    args: [
+      USDC_ADDRESS[chainId],
+      OTOKEN_FACTORY[chainId],
+      GAMMA_CONTROLLER[chainId],
+      MARGIN_POOL[chainId],
+    ],
+    libraries: {
+      VaultLifecycleTreasury: lifecycleTreasury.address,
+    },
+  });
+
+  console.log(`AutocallVault Logic @ ${logicDeployment.address}`);
+
+  try {
+    await run("verify:verify", {
+      address: lifecycleTreasury.address,
+      constructorArguments: [],
+    });
+  } catch (error) {
+    console.log(error);
+  }
+
+  try {
+    await run("verify:verify", {
+      address: logicDeployment.address,
+      constructorArguments: [
+        USDC_ADDRESS[chainId],
+        OTOKEN_FACTORY[chainId],
+        GAMMA_CONTROLLER[chainId],
+        MARGIN_POOL[chainId],
+      ],
+    });
+  } catch (error) {
+    console.log(error);
+  }
+
+  // Autocall vault specific initArgs
+  const initOptionType = 0;
+  const initCouponType = 3;
+  const initAB = 10500;
+  const initNAB = 10500;
+  const initCB = 10500;
+  const initNCB = 10500;
+  const obsFreq = 518400; // 6 days
```

Comment:
can set initNAB / initNCB to 0

Reply:
https://github.com/ribbon-finance/ribbon-v2/pull/313/commits/c9870e9f6646b06c8e618677cca57b82b347312c addressed the above with a small fix.

made a small fix in the .sol `initialize` function to ensure that state variables AB and CB are checked instead of nAB and nCB since the former are the ones that will be used on l120 and l121 of the .sol file.